src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -116,12 +119,131 @@ public static IResourceBuilder<RedisResource> WithRedisCommander(this IResourceB
+                    using var writer = new Utf8JsonWriter(stream);
+                    // Need to grant read access to the config file on unix like systems.
+                    if (!OperatingSystem.IsWindows())
+                    {
+                        File.SetUnixFileMode(connectionFilePath, UnixFileMode.UserRead | UnixFileMode.UserWrite | UnixFileMode.GroupRead | UnixFileMode.OtherRead);
+                    }
+
+                    writer.WriteStartArray();
+
+                    foreach (var redisResource in redisInstances)
+                    {
+                        if (redisResource.PrimaryEndpoint.IsAllocated)
+                        {
+                            var endpoint = redisResource.PrimaryEndpoint;
+                            writer.WriteStartObject();
+                            writer.WriteString("host", endpoint.ContainerHost);
+                            writer.WriteNumber("port", endpoint.Port);
+                            writer.WriteString("name", redisResource.Name);
+                            writer.WriteNumber("db", 0);
+                            //todo: provide username and password when https://github.com/dotnet/aspire/pull/4642 merged.
+                            writer.WriteNull("username");
+                            writer.WriteNull("password");
+                            writer.WriteString("connectionType", "STANDALONE");
+                            writer.WriteEndObject();
+                        }
+                    }
+                    writer.WriteEndArray();
+                };
+
+                var redisInsightResource = builder.ApplicationBuilder.Resources.OfType<RedisInsightResource>().Single();
+
+                var httpClientFactory = e.Services.GetRequiredService<IHttpClientFactory>();
+
+                var client = httpClientFactory.CreateClient();
+
+                var content = new MultipartFormDataContent();
+
+                using (var file = File.OpenRead(connectionFilePath))
+                {
+                    var fileContent = new StreamContent(file);
+
+                    content.Add(fileContent, "file", "RedisInsight_connections.json");
+
+                    var insightEndpoint = redisInsightResource.PrimaryEndpoint;
+                    var apiUrl = $"{insightEndpoint.Scheme}://{insightEndpoint.Host}:{insightEndpoint.Port}/api/databases/import";
+
+                    var pipeline = new ResiliencePipelineBuilder().AddRetry(new Polly.Retry.RetryStrategyOptions
+                    {
+                        Delay = TimeSpan.FromSeconds(2),
+                        MaxRetryAttempts = 5,
+                    }).Build();
+
+                    await pipeline.ExecuteAsync(async (ctx) =>
+                    {
+                        var response = await client.PostAsync(apiUrl, content, ctx)
+                        .ConfigureAwait(false);
+                        response.EnsureSuccessStatusCode();
+
+                    }, ct).ConfigureAwait(false);
```

Comment:
If this fails, it'll stop the entire apphost from loading. I don't think we want that. Should it just log to the output of this resource?